cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java

```diff
@@ -385,14 +389,24 @@ public void testDeployVersionedAndNonVersionedApp() throws Exception {
     Assert.assertEquals(versions, receivedVersions);
 
     appDetails = getAppDetails(appId.getNamespace(), appId.getApplication(), appIdVersion);
-    Assert.assertEquals(GSON.toJson(config), appDetails.getConfiguration());
     Assert.assertEquals(appIdVersion, appDetails.getAppVersion());
+    // when LCM is enabled a new app version is generated else -SNAPSHOT is overwritten
+    if (cConf.getBoolean(FEATURE_FLAG_PREFIX + Feature.LIFECYCLE_MANAGEMENT_EDIT.getFeatureFlagString())) {
+      Assert.assertEquals(GSON.toJson(config), appDetails.getConfiguration());
```

Comment:
I see this was already being done, but it's not good to check equality by checking json strings. For example, ordering of fields in an object is non-deterministic.

Equality should be checked using the proper Java classes